cluster-autoscaler/main.go

```diff
@@ -252,6 +263,23 @@ func createAutoscalingOptions() config.AutoscalingOptions {
 	if *maxDrainParallelismFlag > 1 && !*parallelDrain {
 		klog.Fatalf("Invalid configuration, could not use --max-drain-parallelism > 1 if --parallel-drain is false")
 	}
+
+	// in order to avoid inconsistent deletion thresholds for the legacy planner and the new actuator, the max-empty-bulk-delete,
+	// and max-scale-down-parallelism flags must be set to the same value.
+	if isFlagPassed("max-empty-bulk-delete") && !isFlagPassed("max-scale-down-parallelism") {
+		// Set maxScaleDownParallelismFlag = maxEmptyBulkDeleteFlag to avoid inconsistent deletion thresholds for the legacy planner and the new actuator.
+		*maxScaleDownParallelismFlag = *maxEmptyBulkDeleteFlag
+	} else if !isFlagPassed("max-empty-bulk-delete") && isFlagPassed("max-scale-down-parallelism") {
+		// Set maxEmptyBulkDeleteFlag = maxScaleDownParallelismFlag to avoid inconsistent deletion thresholds for the legacy planner and the new actuator.
+		*maxEmptyBulkDeleteFlag = *maxScaleDownParallelismFlag
+	} else {
```

Comment:
If both flags are passed, we shouldn't emit the warning. No new code should be needed in such case.

Reply:
I agree, I was worried if both flags are set, and they are set to different values, we will still have inconsistent deletion behavior. 


We should just remove the reference to max-empty-bulk-delete https://github.com/kubernetes/autoscaler/pull/5926/files similar to this. I can do this in a separate PR, but this would solve the problem of a user can misconfigure the autoscaler. 


For now, removed the else block, and put a warning message inside of the if statement for max-empty-bulk-delete is set.